### ISSUE TITLE
Disparate Impact Remover doesn't seem to be working

### ISSUE THREAD

**Issue by akuzni2** [Trusted-AI/AIF360#241]
I'm trying to run the disparate impact remover on a dataset - however after varying the repair value between 0 -> 1.0 I see no change in disparate impact metric.

Example:
My dataset (df_encoded) looks like the below with around 600 samples.
![image](https://user-images.githubusercontent.com/25558903/116347259-8194e180-a7b1-11eb-829d-6d6413071775.png)

dataset_train = StandardDataset(df_encoded, label_name='outcome',
                    favorable_classes=[0],
                    protected_attribute_names=['sensitive'],
                    privileged_classes=[[1]])

di_remover = DisparateImpactRemover(repair_level=1.0)
train_repd = di_remover.fit_transform(dataset_train)

privileged_groups = [{'sensitive': 1}]
unprivileged_groups = [{'sensitive': 0}]
    
metric_train_repd = BinaryLabelDatasetMetric(train_repd, 
                unprivileged_groups=unprivileged_groups,
                privileged_groups=privileged_groups)

explainer = MetricTextExplainer(metric_train_repd)
explainer.disparate_impact()

In my dataset i get a disparate impact of 0.71. When I change "repair_level" to 0 or 1.0 I see no change in disparate impact.

**Comment by cjwillis48** [comment]
Was a solution found for this? I am seeing the same thing now

**Comment by OnTheThirdDay** [comment]
I think that is actually the intended result. You may compare the dataset before and after, and the disparate_impact_remover has adjusted the feature values.

**Comment by pradeepdev-1995** [comment]
@OnTheThirdDay How can I compare the datasets( original dataset and the disparate imapact removed dataset) through code?

**Comment by OnTheThirdDay** [comment]
> @OnTheThirdDay How can I compare the datasets( original dataset and the disparate impact removed dataset) through code?

Hi @pradeepdev-1995 , I haven't followed this thread for long, so a bit forgot the context.
To compare, you can export the dataset and you can see the feature values are actually different, though the DI score is not changed.

**Comment by pradeepdev-1995** [comment]
```
di_remover = DisparateImpactRemover(repair_level=1.0)
train_repd = di_remover.fit_transform(dataset_train)
```
How can I export the ```train_repd``` to csv?
kindly share the code snippet for this.

**Comment by OnTheThirdDay** [comment]
> ```
> di_remover = DisparateImpactRemover(repair_level=1.0)
> train_repd = di_remover.fit_transform(dataset_train)
> ```
> 
> How can I export the `train_repd` to csv? kindly share the code snippet for this.

Use some api like .convert_to_dataframe() and to_csv()

**Comment by pradeepdev-1995** [comment]
@OnTheThirdDay 
Checked with ```convert_to_dataframe``` function of AIF360 like this
```
di = DisparateImpactRemover(repair_level=1.0)
dir_dataset = di.fit_transform(dataset)
dir_dataset_df= dir_dataset.convert_to_dataframe(sep=",")

print(originalDataset.equals(dir_dataset_df[0]))
```
But it shows that original and transformed datasets are the same. Which means DisparateImpactRemover nothing changed in the dataset even the repair_level=1.0

**Comment by alhasan-ramadan** [comment]
Is this still a thing? Using the Adult Income Dataset, the DIR doesnt seem to work when setting male/female and targeting income
